mindmeld/models/taggers/pytorch_crf.py

```diff
@@ -0,0 +1,408 @@
+import logging
+import os
+import random
+import uuid
+from collections import Counter
+from copy import copy
+from itertools import zip_longest
+from random import randint
+
+import numpy as np
+import torch
+import torch.nn as nn
+from sklearn.feature_extraction import DictVectorizer, FeatureHasher
+from sklearn.metrics import f1_score
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import LabelEncoder
+from torch import optim
+from torch.utils.data import Dataset, DataLoader
+from torchcrf import CRF
+
+from ...exceptions import MindMeldError
+from ...path import USER_CONFIG_DIR
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_PYTORCH_CRF_ER_CONFIG = {
+    "feat_type": "hash",  # ["hash", "dict"]
+    "feat_num": 50000,
+    "stratify_train_val_split": True,
+    "drop_input": 0.2,
+    "train_batch_size": 8,
+    "patience": 3,
+    "epochs": 100,
+    "train_val_split": 0.15,
+    "optimizer_type": "sgd",  # ["sgd", "adam"]
+}
```

Comment:
Would like to bring up the [comment](https://github.com/cisco/mindmeld/pull/413#discussion_r882145224) about trying to maintain similar terminology across different modules that use pytorch.

Reply:
Renamed to be consistent across NN models